src/posit/connect/resources.py

```diff
@@ -43,3 +46,254 @@ def update(self, *args, **kwargs):
 class Resources:
     def __init__(self, params: ResourceParameters) -> None:
         self.params = params
+
+
+class Active(Resource):
+    """
+    A base class representing an active resource.
+
+    Extends the `Resource` class and provides additional functionality for via the session context and an optional parent resource.
+
+    Parameters
+    ----------
+    ctx : Context
+        The context object containing the session and URL for API interactions.
+    parent : Optional[Active], optional
+        An optional parent resource that establishes a hierarchical relationship, by default None.
+    **kwargs : dict
+        Additional keyword arguments passed to the parent `Resource` class.
+
+    Attributes
+    ----------
+    _ctx : Context
+        The session context.
+    _parent : Optional[Active]
+        The parent resource, if provided, which establishes a hierarchical relationship.
+    """
+
+    def __init__(self, ctx: Context, parent: Optional["Active"] = None, **kwargs):
+        """
+        Initialize the `Active` resource.
+
+        Parameters
+        ----------
+        ctx : Context
+            The context object containing session and URL for API interactions.
+        parent : Optional[Active], optional
+            An optional parent resource to establish a hierarchical relationship, by default None.
+        **kwargs : dict
+            Additional keyword arguments passed to the parent `Resource` class.
+        """
+        params = ResourceParameters(ctx.session, ctx.url)
+        super().__init__(params, **kwargs)
+        self._ctx = ctx
+        self._parent = parent
+
+
+T_co = TypeVar("T_co", bound="Active", covariant=True)
+"""A covariant type variable that is bound to the `Active` class, meaning that `T_co` must be or derive from `Active`."""
```

Comment:
What is the purpose of this restriction?

Reply:
After a discussion with @schloerke, we realized this doesn't need to be covariant. I've updated the usage to `TypeVar("T", bound="Active")`